dashboard/test/controllers/foorm/library_questions_controller_test.rb

```diff
@@ -35,5 +41,66 @@ class LibraryQuestionsControllerTest < ActionController::TestCase
 
       assert_response :success
     end
+
+    test 'create succeeds for new library question in existing library' do
+      levelbuilder = create :levelbuilder
+      sign_in levelbuilder
+
+      library = create :foorm_library
+
+      post :create, params: {
+        library_id: library.id,
+        question: {pages: [{elements: [{name: "test"}]}]},
+        name: 'test_question_name'
+      }
+
+      # Timestamp format is different if you do not as_json ActiveRecord objects
+      assert_equal Hash(
+        library_question: Foorm::LibraryQuestion.where(question_name: 'test_question_name').first.as_json,
+        library: library.as_json
+      ).as_json, JSON.parse(response.body)
+    end
+
+    test 'create succeeds for new library question in new library' do
+      levelbuilder = create :levelbuilder
+      sign_in levelbuilder
+
+      post :create, params: {
+        library_name: 'test_library_name',
+        question: {pages: [{elements: [{name: "test"}]}]},
+        name: 'test_question_name'
+      }
+
+      assert_equal Hash(
```

Comment:
Looks like we have a `test_helpers` with `assert_create` and other helpful methods you may be able to leverage

Reply:
It looks like `assert_creates` checks that a new object is created given some request -- I use a (questionable) approach here that ends up returning both the library and library question when a library question is created, so I think I need some sort of custom check to make sure things are returned properly. 

The whole thing is kind of a hack because we don't have the Rails library ID in the library question model (just the library name and version), and I need the actual Rails library ID to keep track of what libraries exist in the UI.